pkg/apis/experimental/types.go

```diff
@@ -461,3 +461,75 @@ type JobCondition struct {
 	// Human readable message indicating details about last transition.
 	Message string `json:"message,omitempty"`
 }
+
+// IngressPoint encapsulates the inputs needed to connect an alias to endpoints.
+// It represents a rule collection of inbound connections from the external network
+// that would be satisfied by a load balancer.
+type IngressPoint struct {
+	api.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	api.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the desired state of the IngressPoint.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+	Spec IngressPointSpec `json:"spec,omitempty"`
+
+	// Status is the current state of the IngressPoint.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+	Status IngressPointStatus `json:"status,omitempty"`
+}
+
+// IngressPointList is a collection of IngressPoints,
+type IngressPointList struct {
+	api.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	api.ListMeta `json:"metadata,omitempty"`
+
+	// Items is the list of ingressPoints.
+	Items []IngressPoint `json:"items"`
+}
+
+// ServiceRef represents a reference to a single service:port.
+type ServiceRef struct {
+	// Define the name of the referenced service.
+	Name string `json:"name,omitempty"`
+
+	// Define the namespace of the referenced service.
+	Namespace string `json:"namespace,omitempty"`
+
+	// Define the service's port which the http traffic will be forwarded to.
+	Port int `json:"port,omitempty"`
+}
+
+// PathRef represents a single load balancer rule of forwarding the specified url to the related backend service.
+type PathRef struct {
+	// Domain describe http url's domain.
+	// Examples: "foo.example.com", "bar.example.com".
+	Domain string `json:"domain,omitempty"`
```

Comment:
Actually this is the subdomain, in eg: maps.google.com this is maps, domain is google and tld is .com. A single ingresspoint can only have a single domain.tld, but it can have multiple subdomains. 

We can have multiple paths+serviceref for a single domain, i mean:
maps.google.com/foo->svc1, maps.google.com/bar->svc2 ...
So this needs to contain a subobject which is a list.

```
PathRef {
    SubDomain: string,
    Path: []Path,
}

Path {
    Url (need a better name for this): string,
    Service: ServiceRef,
}
```

@ramr I think had a better name for url in the L7 proposal